crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -697,7 +701,28 @@ where
 {
     ignore_then_commit(keyword(Keyword::Assert), parenthesized(expr_parser))
         .labelled(ParsingRuleLabel::Statement)
-        .map(|expr| Statement::Constrain(ConstrainStatement(expr)))
+        .map(|expr| {
+            let true_literal =
+                Expression::new(ExpressionKind::Literal(Literal::Bool(true)), expr.span);
+            Statement::Constrain(ConstrainStatement(expr, true_literal))
+        })
+}
+
+fn assertion_eq<'a, P>(expr_parser: P) -> impl NoirParser<Statement> + 'a
+where
+    P: ExprParser + 'a,
+{
+    ignore_then_commit(
+        keyword(Keyword::AssertEq),
+        parenthesized(expr_parser.separated_by(just(Token::Comma)).exactly(2)),
+    )
+    .labelled(ParsingRuleLabel::Statement)
+    .validate(|exprs: Vec<Expression>, span, _| {
```

Comment:
Legitimate question: Is this type expression needed? Not sure if Rust would infer this or not due to the trait indirection here.

Reply:
Yeah, I get an `{unknown}` showing up if I don't have this.